src/brain.coffee

```diff
@@ -997,38 +996,59 @@ class Brain
     # Get the substitutions map.
     subs = if type is "sub" then @master._sub else @master._person
 
-    # Make placeholders each time we substitute something.
-    ph = []
-    pi = 0
-
-    for pattern in @master._sorted[type]
-      result = subs[pattern]
-      qm     = utils.quotemeta pattern
-
-      # Make a placeholder.
-      ph.push result
-      placeholder = "\x00<#{pi}>\x00"
-      pi++
+    # Get the max number of words in sub/person to minimize interations
+    maxwords = if type is "sub" then @master._submax else @master._personmax
 
-      # Run substitutions.
-      msg = msg.replace(new RegExp("^#{qm}$", "g"),           placeholder)
-      msg = msg.replace(new RegExp("^#{qm}([^a-zA-Z0-9_>]+)", "g"),      "#{placeholder}$1")
-      msg = msg.replace(new RegExp("([^a-zA-Z0-9_<]+)#{qm}([^a-zA-Z0-9_>]+)", "g"), "$1#{placeholder}$2")
-      msg = msg.replace(new RegExp("([^a-zA-Z0-9_<]+)#{qm}$", "g"),      "$1#{placeholder}")
+    result = ""
 
-    # Convert the placeholders back in.
+    # Take the original message with no punctuation
+    pattern = msg.replace(/[.?,!;:]/, "")
```

Comment:
This could probably reuse the `@master.unicodePunctuation` regexp, and that way if UTF-8 supporting users change the set of punctuation symbols to fit their target language it will apply here too.